state/execution.go

```diff
@@ -116,6 +116,17 @@ func (blockExec *BlockExecutor) CreateProposalBlock(
 		panic(err)
 	}
 
+	appHash := state.AppHash // adapt abci to abcix, appHash is received from executing previous block
+	if resp.AppHash != nil {
```

Comment:
don't compare `[]byte` with nil, because `[]byte{}` will not match but usually is not expected as well.

but why do we need to care about length of `AppHash`? shouldn't we always trust `resp.AppHash` no matter it's empty or not?

Reply:
actually, this is part of adapter, if it's an `adapt abci app`, its `appHash` should be `state.AppHash`, however, I can't set the value in `adapter.go`

Reply:
you will do something like this in this PR #56, where adapter will keep track of last app hash from commit. because `state.AppHash` comes from app's `Commit`